core/templates/dev/head/i18n.js

```diff
@@ -45,20 +45,26 @@ oppia.constant('DEFAULT_TRANSLATIONS', {
 });
 
 oppia.controller('I18nFooter', [
-    '$rootScope', '$scope', '$translate',
-    function($rootScope, $scope, $translate) {
+    '$http', '$rootScope', '$scope', '$translate',
+    function($http, $rootScope, $scope, $translate) {
   // Changes the language of the translations.
+  var _PREFERENCES_DATA_URL = '/preferenceshandler/data';
+  var _SAVE_SITE_LANGUAGE_URL = '/save_site_language';
   $scope.supportedSiteLanguages = GLOBALS.SUPPORTED_SITE_LANGUAGES;
+  if (GLOBALS.userIsLoggedIn) {
+    $http.get(_PREFERENCES_DATA_URL).success(function(data) {
+      $translate.use(data.preferred_site_language_code);
```

Comment:
Oh now I see what you mean! Actually now it's only making a single request the first time the user access the Oppia site and is logged in, which is less request than setting a global in base.py. I'm really busy settling in a new job and a new location, I'm not going to be able to fix this prior to weekend. If you want, you can merge as is because it's working and open a new issue for the later modification, or wait until Sunday. I don't want to delay more the merging with develop. Sorry about that!